org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IBuildSupport.java

```diff
@@ -70,13 +72,24 @@ default void refresh(IResource resource, CHANGE_TYPE changeType, IProgressMonito
 			return;
 		}
 		if (changeType == CHANGE_TYPE.DELETED) {
+			if (IJavaProject.CLASSPATH_FILE_NAME.equals(resource.getName())) {
+				IProject project = resource.getProject();
+				if (ProjectUtils.isJavaProject(project)) {
+					ProjectUtils.removeJavaNatureAndBuilder(project, monitor);
+					updateBuildFile(project);
```

Comment:
that's too hacky. should directly call project update instead

Reply:
Fixed.

> Need to check if CLASSPATH_FILE_NAME is at the root of its project (else it might be harmless)....

The project update method recreates CLASSPATH_FILE_NAME automatically.